server.lua

```diff
@@ -187,33 +187,33 @@ RegisterNetEvent('qb-vehicleshop:server:financePaymentFull', function(data)
     end
 end)
 
+---Checks if player has enough money, then returns a string based on the currency.
+---@param price number
+---@param cash number
+---@param bank number
+---@return string
+local function findChargeableCurrencyType(price, cash, bank)
+    if cash >= price then
+        return 'cash'
+    elseif bank >= price then
+        return 'bank'
+    else
+        return nil
+    end
+end
+
 -- Buy public vehicle outright
 RegisterNetEvent('qb-vehicleshop:server:buyShowroomVehicle', function(vehicle)
     local src = source
-    vehicle = vehicle.buyVehicle
-    local pData = QBCore.Functions.GetPlayer(src)
-    local cid = pData.PlayerData.citizenid
-    local cash = pData.PlayerData.money['cash']
-    local bank = pData.PlayerData.money['bank']
+    local player = QBCore.Functions.GetPlayer(src)
     local vehiclePrice = QBCore.Shared.Vehicles[vehicle]['price']
-    local plate = GeneratePlate()
-    if cash > tonumber(vehiclePrice) then
-        MySQL.insert('INSERT INTO player_vehicles (license, citizenid, vehicle, hash, mods, plate, garage, state) VALUES (?, ?, ?, ?, ?, ?, ?, ?)', {
-            pData.PlayerData.license,
-            cid,
-            vehicle,
-            GetHashKey(vehicle),
-            '{}',
-            plate,
-            'pillboxgarage',
-            0
-        })
-        TriggerClientEvent('QBCore:Notify', src, Lang:t('success.purchased'), 'success')
-        TriggerClientEvent('qb-vehicleshop:client:buyShowroomVehicle', src, vehicle, plate)
-        pData.Functions.RemoveMoney('cash', vehiclePrice, 'vehicle-bought-in-showroom')
-    elseif bank > tonumber(vehiclePrice) then
+    local currencyType = findChargeableCurrencyType(vehiclePrice, player.PlayerData.money['cash'], player.PlayerData.money['bank'])
```

Comment:
```suggestion
    local currencyType = findChargeableCurrencyType(vehiclePrice, player.PlayerData.money.cash, player.PlayerData.money.bank)
```

https://manason.github.io/effective-fivem-lua/tables/#prefer-object-access-for-constant-keys-and-array-access-for-non-constant-keys